app/javascript/components/mentoring/mentor_conversations_list.jsx

```diff
@@ -0,0 +1,106 @@
+import React, { useState } from 'react'
+import { usePaginatedQuery } from 'react-query'
+import dayjs from 'dayjs'
+import RelativeTime from 'dayjs/plugin/relativeTime'
+import { Pagination } from './mentor_conversations_list/pagination'
+import { UrlParams } from '../../utils/url_params'
+dayjs.extend(RelativeTime)
+
+function MentorConversationListRow({
+  trackTitle,
+  trackIconUrl,
+  menteeAvatarUrl,
+  menteeHandle,
+  exerciseTitle,
+  isStarred,
+  haveMentoredPreviously,
+  isNewIteration,
+  postsCount,
+  updatedAt,
+  url,
+}) {
+  return (
+    <tr>
+      <td>
+        <img
+          style={{ width: 100 }}
+          src={trackIconUrl}
+          alt={`icon indicating ${trackTitle}`}
+        />
+      </td>
+      <td>
+        <img
+          style={{ width: 100 }}
+          src={menteeAvatarUrl}
+          alt={`avatar for ${menteeHandle}`}
+        />
+      </td>
+      <td>{menteeHandle}</td>
+      <td>{exerciseTitle}</td>
+      <td>{isStarred.toString()}</td>
+      <td>{haveMentoredPreviously.toString()}</td>
+      <td>{isNewIteration.toString()}</td>
+      <td>{postsCount}</td>
+      <td>{dayjs(updatedAt).fromNow()}</td>
+      <td>{url}</td>
+    </tr>
+  )
+}
+
+async function fetchSolutions(key, url, params) {
+  const resp = await fetch(url)
+
+  return resp.json()
+}
+
+export function MentorConversationsList({ endpoint, ...props }) {
+  const [page, setPage] = useState(1)
+  const urlParams = new UrlParams(
+    Object.assign({ page: page }, props.urlParams)
+  )
+  const retryParams = props.retryParams
+  const { status, resolvedData, latestData } = usePaginatedQuery(
+    ['mentor-conversations-list', `${endpoint}?${urlParams.toString()}`],
+    fetchSolutions,
+    retryParams
+  )
+
+  return (
+    <div>
+      {status === 'loading' && <p>Loading</p>}
+      {status === 'error' && <p>Something went wrong</p>}
```

Comment:
Should we add a retry button if we get this? (Not for this PR, but as a future thing)

Reply:
Yeah. Good idea. FYI, our react library already retries 3 times before showing the error state. Still a good idea to add a retry button though.